addons/purchase/models/purchase.py

```diff
@@ -474,7 +474,8 @@ def _add_supplier_to_product(self):
         for line in self.order_line:
             # Do not add a contact as a supplier
             partner = self.partner_id if not self.partner_id.parent_id else self.partner_id.parent_id
-            if line.product_id and partner not in line.product_id.seller_ids.mapped('name') and len(line.product_id.seller_ids) <= 10:
+            already_seller = (partner | self.partner_id) & line.product_id.seller_ids.mapped('name')
+            if line.product_id and not already_seller and partner not in line.product_id.seller_ids.mapped('name') and len(line.product_id.seller_ids) <= 10:
```

Comment:
the condition above already checks `partner not in line.product_id.seller_ids.mapped('name')`

```suggestion
            if line.product_id and not already_seller and len(line.product_id.seller_ids) <= 10:
```